index/scorch/persister.go

```diff
@@ -871,24 +886,79 @@ func (s *Scorch) removeOldData() {
 // rollback'ability.
 var NumSnapshotsToKeep = 1
 
+// RollbackSamplingInterval controls how far back we are looking
+// in the history to get the rollback points.
+// For example, the default value of 1 hour ensures that the
+// protected snapshots (NumSnapshotsToKeep = 3) are:
+//    the very latest snapshot(ie the current one),
+//    the snapshot that was persisted 1 hour before the current one,
+//	  the snapshot that was persisted 2 hours before the current one
+var RollbackSamplingInterval = 1 * time.Hour
+
+// getProtectedEpochs aims to fetch the epochs keep based on a timestamp basis.
+// It tries to get NumSnapshotsToKeep snapshots, each of which are separated
+// by a time duration of RollbackSamplingInterval.
+func (s *Scorch) getProtectedEpochs(
+	persistedSnapshots []*snapshotMetaData) map[uint64]struct{} {
+
+	// make a map of epochs to protect from deletion
+	protectedEpochs := make(map[uint64]struct{}, s.numSnapshotsToKeep)
```

Comment:
I don't think providing a length parameter while "make"ing over a map does anything does it? Just this would suffice ..

    protectedEpochs := make(map[uint64]struct{})

Reply:
Done.